src/CoreWCF.Http/src/CoreWCF/Channels/AspNetCoreReplyChannel.cs

```diff
@@ -120,6 +120,12 @@ internal async Task HandleRequestCore(HttpContext context)
             }
 
             var requestContext = HttpRequestContext.CreateContext(_httpSettings, context);
+            bool isAuthenticated = await requestContext.ProcessAuthenticationAsync();
+            if (!isAuthenticated)
+            {
+                // throw exception?
+            }
```

Comment:
I'm not seeing a way for ProcessAuthentication to return false. It either returns true or throws an exception. This is because ValidateAuthentication always returns OK. I need to spend a bit more time looking at the original WCF code and work out what the equivalent conditions are, but I think there's enough feedback for you to be getting on with for now.